osu.Server.Spectator/Hubs/MultiplayerHub.cs

```diff
@@ -574,6 +574,8 @@ public async Task ChangeSettings(MultiplayerRoomSettings settings)
 
                 if (previousSettings.AutoStartDuration != settings.AutoStartDuration)
                 {
+                    // If autostart is still enabled, the countdown will be restarted in the `updateRoomStateIfRequired` call below.
+                    // Stopping the countdown here is important to ensure the new duration is applied over any previous autostart (or manual) countdown.
```

Comment:
It won't be restarted because all users are unreadied.

Reply:
Ah, that's weird. Are we reading different PR descriptions?

Let me re-review this.

Reply:
If that's the case, I think it may be more correct to put this call alongside the unready-all-users call? I'd probably expect any active countdown to restart if the host suddenly changes the match settings.

Reply:
_Any_ match settings? Changing the room name should also cause the countdown to be cancelled? The other case that I might agree with is when the current playlist item changes, which also calls `unreadyAllUsers`, but I don't know about blanketing it on all match settings changes.

Reply:
Fair, how about a `bool` argument to make it up to the caller?

Reply:
I guess you could apply the same thought behind calling `unreadyAllUsers` in the first place - should all users be unreadied even when it's only the room name that changes? Probably not.

Will move to `unreadyAllUsers`.

Reply:
That's fine for now, but I think `NotifyMatchSettingsChanged(MultiplayerRoom room, bool resetReadyStates = true)` might work quite well.

Reply:
Sure, but I'll leave that to a separate change because there's other flags involved and more tests to be made.